python/GafferSceneUI/_SceneViewInspector.py

```diff
@@ -0,0 +1,959 @@
+			if plug is not None :
+				return _EditInfo(
+					acquireEdit = lambda : plug,
+					editType = _EditType.Source,
+					inEditScope = node.ancestor( Gaffer.EditScope ) is not None,
+					hasEdit = True
+				)
+			return None
+
+		elif isinstance( node, GafferScene.ShaderAssignment ) :
+
+			with attributeHistory.context :
+
+				if not node["filter"].getValue() & IECore.PathMatcher.Result.ExactMatch :
+					return None
+
+				shaderInput = node["shader"].getInput()
+				if shaderInput is not None :
+					plug = shaderInput.node()["parameters"].getChild( self.__parameter )
+					if plug is not None :
+						return _EditInfo(
+							acquireEdit = lambda : plug,
+							editType = _EditType.Source,
+							inEditScope = node.ancestor( Gaffer.EditScope ) is not None,
+							hasEdit = True
+						)
+				return None
+
+		elif isinstance( node, GafferScene.ShaderTweaks ) :
+
+			with attributeHistory.context :
+
+				if not node["filter"].getValue() & IECore.PathMatcher.Result.ExactMatch :
+					return None
+
+				for tweak in node["tweaks"] :
+					if tweak["name"].getValue() == self.__parameter :
+						# As we adopt the enabled plug, this will always fail
+						# for Tweaks in a spreadsheet as the adopted cell
+						# enabled state points us to the default row.
+						tweak = self.__spreadsheetAwareSource( tweak )
+						if tweak is not None :
+							return _EditInfo(
+								acquireEdit = lambda : tweak,
+								editType = _EditType.Tweak,
+								inEditScope = node.ancestor( Gaffer.EditScope ) is not None,
+								hasEdit = tweak["enabled"].getValue()
+							)
+						return None
+
+		return None
+
+	def __editFromEditScope( self, attributeHistory ) :
+
+		def editScopeEdit( attributeHistory, parameter ) :
+
+			with attributeHistory.context :
+				return GafferScene.EditScopeAlgo.acquireParameterEdit(
+					attributeHistory.scene.node(),
+					attributeHistory.context["scene:path"],
+					attributeHistory.attributeName,
+					IECoreScene.ShaderNetwork.Parameter( "", parameter )
+				)
+
+		return _EditInfo(
+			acquireEdit = functools.partial( editScopeEdit, attributeHistory, self.__parameter ),
+			editType = _EditType.Tweak,
+			inEditScope = True,
+			hasEdit = self.__editScopeHasEdit( attributeHistory )
+		)
+
+	def __editScopeHasEdit( self, attributeHistory ) :
+
+		with attributeHistory.context :
+
+			tweak = GafferScene.EditScopeAlgo.acquireParameterEdit(
+				attributeHistory.scene.node(),
+				attributeHistory.context["scene:path"],
+				attributeHistory.attributeName,
+				IECoreScene.ShaderNetwork.Parameter( "", self.__parameter ),
+				createIfNecessary = False
+			)
+
+			if tweak is None :
+				return False
+
+			return tweak["enabled"].getValue()
+
+	def __findEdit( self, attributeHistory, editScope ) :
+
+		# Walk through the attributeHistory, looking for suitable edit, based on
+		# the users chosen editScope
+
+		if editScope is not None :
```

Comment:
What's the reason for putting this check here at the start, instead of with the `readOnly()` check we perform when we actually find the edit scope? If we perform the check at the point we find the node, we will be able to use the correct context to evaluate "enabled" in.

Reply:
Improved [here](https://github.com/GafferHQ/gaffer/pull/3832/commits/5c56b87e41f1ab42b3f73054f92a76006970de6a).

Reply:
Additional checks added [here](https://github.com/GafferHQ/gaffer/pull/3832/commits/502716903d5da732d6325409cecfb8e81a5866a6) too.